packages/hebao_v1/contracts/modules/core/ForwarderModule.sol

```diff
@@ -26,15 +26,14 @@ abstract contract ForwarderModule is BaseModule
     uint    public constant MAX_REIMBURSTMENT_OVERHEAD = 165000;
     bytes32 public FORWARDER_DOMAIN_SEPARATOR;
 
+    mapping(address => uint) public nonces;
```

Comment:
I think all the data should be stored in some DataStore for future upgrading?

Reply:
Yes, but nonces aren't especially something that need to be kept between upgrades. If the user updates the forwarder module the nonces restart from `0` (with a different `FORWARDER_DOMAIN_SEPARATOR` value) which should be okay. The user doesn't care about the nonce value, unlike things like guardians, quota, etc...

Reply:
OK, so we should set a different value to FORWARDER_DOMAIN_SEPARATOR when we upgrading the ForwarderModule. Otherwise a replay-attach becomes possible.

Reply:
Done automatically because of EIP712, the contract address is always part of the data hashed to FORWARDER_DOMAIN_SEPARATOR.